lib/Util/Debug.cpp

```diff
@@ -0,0 +1,86 @@
+//===-- Debug.cpp -----------------------------------------------*- C++ -*-===//
+//
+// Part of the LLZK Project, under the Apache License v2.0.
+// See LICENSE.txt for license information.
+// Copyright 2025 Veridise Inc.
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#include "llzk/Util/Debug.h"
+#include "llzk/Util/StreamHelper.h"
+
+using namespace mlir;
+
+namespace llzk {
+namespace debug {
+
+void dumpSymbolTableWalk(Operation *symbolTableOp) {
+  std::string output; // buffer to avoid multi-threaded mess
+  llvm::raw_string_ostream oss(output);
+  oss << "Dumping symbol walk (self = [" << symbolTableOp << "]): \n";
+  auto walkFn = [&](Operation *op, bool allUsesVisible) {
+    oss << "  found op [" << op << "] " << op->getName() << " named "
+        << op->getAttrOfType<StringAttr>(SymbolTable::getSymbolAttrName()) << '\n';
+  };
+  SymbolTable::walkSymbolTables(symbolTableOp, /*allSymUsesVisible=*/true, walkFn);
+  llvm::outs() << output;
+}
+
+void dumpSymbolTable(llvm::raw_ostream &stream, SymbolTable &symTab, unsigned indent) {
+  for (unsigned i = 0; i < indent; ++i) {
+    stream << "  ";
+  }
```

Comment:
```suggestion
  stream.indent(indent);
```
https://llvm.org/doxygen/classllvm_1_1raw__ostream.html#a8fdf5cdf041c8aded7e3308c1c3efacc

Reply:
nice, didn't know that existed